docs/source-pytorch/advanced/model_parallel/fsdp.rst

```diff
@@ -2,138 +2,439 @@
-    from torch.distributed.fsdp.wrap import wrap
+    # 1. Define a set of layers that FSDP should manage
+    #    Here we are choosing the large encoder and decoder layers
+    policy = {nn.TransformerEncoderLayer, nn.TransformerDecoderLayer}
+
+    # 2. Pass the policy to the FSDPStrategy object
+    strategy = FSDPStrategy(auto_wrap_policy=policy)
+
+    trainer = L.Trainer(..., strategy=strategy)
+
+.. collapse:: Alternative ways to define the policy (Lightning < 2.1)
+
+    The ``auto_wrap_policy`` argument also accepts the old-style function-policies. For example:
+
+    .. code-block:: python
+
+        from functools import partial
+
+        # 1. Import a suiting wrapping policy from PyTorch
+        from torch.distributed.fsdp.wrap import size_based_auto_wrap_policy
+
+        # 2. Configure the policy
+        policy = partial(size_based_auto_wrap_policy, min_num_params=10000)
+
+        # 3. Pass it to the FSDPStrategy object
+        strategy = FSDPStrategy(auto_wrap_policy=policy)
+
+    PyTorch provides several of these functional policies under :mod:`torch.distributed.fsdp.wrap`.
+
+|
+
+Verify that FSDP works with your model by comparing the peak memory usage printed in the CUDA memory summary (see example above) with regular DDP training.
+You should see a decrease in allocated memory and a slight increase in iteration time:
+
+.. list-table::
+   :widths: 25 25 25
+   :header-rows: 1
+
+   * -
+     - DDP
+     - FSDP
+   * - Memory (MB)
+     - 23’125
+     - 9’627
+   * - Iterations per second
+     - 4.31
+     - 3.19
+
+----
 
 
-    class MyModel(pl.LightningModule):
+*****************************
+Speed up model initialization
+*****************************
+
+The standard practice in PyTorch is to put all model parameters into CPU memory first and then in a second step move them to the GPU device.
+However, the larger the model the longer these two steps take.
+If you create the large model layers inside the :meth:`~lightning.pytorch.core.hooks.ModelHooks.configure_model` hook, you can initialize very large models quickly and reduce memory peaks.
+
+Before:
+
+.. code-block:: python
+
+    # Slow: Places the model on CPU first
+    class LanguageModel(L.LightningModule):
+        def __init__(self, vocab_size):
+            super().__init__()
+            self.model = Transformer(vocab_size=vocab_size, nlayers=32, nhid=4096, ninp=1024, nhead=64)  # 1B parameters
+
+After:
+
+.. code-block:: python
+
+    # Fast: Delays the model creation until Trainer can place it on GPU
+    class LanguageModel(L.LightningModule):
+        def __init__(self, vocab_size):
+            super().__init__()
+            self.vocab_size = vocab_size
+            self.model = None
+
         def configure_model(self):
-            self.linear_layer = nn.Linear(32, 32)
-            self.block = nn.Sequential(nn.Linear(32, 32), nn.Linear(32, 32))
+            self.model = self.model or Transformer(  # 1B parameters
```

Comment:
I would advise against using the truthiness of the model for this. You could instead do `if self.model is None: return`